Create 2021-11-15.md (#200)
solid/authentication-panel: meetings/2021-11-15.md

* AC: currently it tests OP client interaction
  * ...: it does test some OIDC interaction, well-known, public key, etc.
  * ...: I'm only testing ID Token, anticipating changes coming next
* eP: noticing annotations, there is nothing stopping us using JSON-LD in the spec document (rather than Turtle). What is the process of extracting the 'statement'.
* AC: the current test suit has no dependencies. I'm pulling descriptions as JSON, since they are in compacted JSON-LD.
  * ...: I have multiple suites. The test is just few lines of code. We pass identifier for relevant test, and whether it passed, plus optional message.
  * ...: In data directory, I can describe new test or requirement, and reference it from test suite.
  * ...: I'm performing some reflection to get methods of the class, so if you add a method to the class, it automatically runs.
* PE: When we interpret the spec, you can see requirement level and requirement subject. If you do that in turtle fragments you would have duplication.
  * ...: Is you report function going directly to HTML? I assume you can start outputting EARL.
* AC: I currently do some DOM manipulation, but we could output the data in any other way.

> * PE: ??? [completely missed comment should probably be deleted, unless @edwardsph remembers what they said here]

Author: ```suggestion
```

We can wait another day before deleting it, in case @edwardsph can recall what is missing here.

Reviewer: Sorry - I can't remember what this was related to but there is nothing obviously missing so I'm happy for that to be deleted.

* AC: In test suite, some requirement statements are pointing to different specs. We can't expect OIDC spec to include RDF with requirements. We can either include requirements for those specs or not test things not defined in OIDC spec.
  * ...: I think it's helpful for developers to test dependencies from OIDC specs. For example, test whether issuer matches.
  * ...: I don't know how far we can go with expecting data to be automatically loaded from the spec. Sometimes we need to have a separate manifest.
* PE: I see value for the developer of tests guiding them.

> * eP: ??? [completely missed comment should probably be deleted, unless @elf-pavlik remembers what they said here]

Author: ```suggestion
* eP: To extract statements from the spec test,  we could make each requirement an element and just use text in that element as a statement. We would still just need a simple id for each requirement.
```

> * AC: There is repository https://github.com/solid/solid-oidc-tests
> * ...: It uses github pages to produce a static app
> * ...: I can for example test Solid OIDC issuer. Then it provides conformance report explaining which tests are passing and which are failing. It also provides MUST / MAY level.
> * ...: It would be nice to get results as RDF. All the tests are described as RDF already.
> * ...: Pete had suggestion, I use comment now, but spec vocab (currently on branch not published) has property called statement.
> * ...: There are some things hard coded, for example using specific crypt algorithms. CSS team has run in against their server, it was only failing very recent change to the spec introducing webid scope.
> * ...: Some UI design could happen here, for now it is a decent start.
> * eP: does it test interaction of OP with a client or also with RS
> * AC: currently it tests OP client interaction
> * ...: it does test some OIDC interaction, well-know, public key etc.
> * ...: I'm only testing ID Token, anticipating changes coming next
> * eP: noticing annotations, there is nothing stopping us using JSON-LD in the spec document (rather than Turtle). What is the process of extracting the 'statement'.
> * AC: the current test suit has no dependencies, I'm pulling descriptions as JSON since they are in compacted JSON-LD.
> * ...: I have multiple suites. The test is just few lines of code. We pass identifier for relevant test and if it passed, plus optional message.
> * ...: In data directory I can describe new test / requirement and reference it from test suite.
> * ...: I'm performing some reflection to get methods of the class, so if you add method to the class if it automatically run.
> * PE: When we interpret the spec, you can see requirement level and requirement subject. If you do that in turtle fragments you would have duplication.
> * ...: Is you report function going directly to HTML? I assume you can start outputting EARL.
> * AC: I currently do some DOM manipulation but we could output the data in any other way.
> * PE:
> * AC: In test suite some requirement statements are pointing to different specs. We can't expect OIDC spec to include RDF with requirements. We can either include requirements for those specs or not test things not defined in OIDC spec.
> * ...: I think it's helpful for developers to test dependencies from OIDC specs. For example test weather issuer matches.
> * ...: I don't know how far we can go with expecting data automatically being loaded from the spec. Sometimes we need to have separate manifest.
> * PE: I see value for the developer of tests guiding them.
> * eP: ...
> * AC: With OIDC we point to the same IRI, we have similar issue with Solid-OIDC. For example, we have basic assumption that we use PKCE and Authorization Code flow but is not normative statement. In my test MUST is not reflecting exactly what the spec test says. We should adjust spec text to reflect those requirements.
> * ...: For discovery we point to the same IRI when we point to webid scope , and supported claims.
> * AC: I have all of the code written as ESM, I'm receptive to suggestions for how to make code better.

Reviewer 2: Indents make it clearer when speaker changes.  There are other suggested changes inline.
```suggestion
* AC: There is repository https://github.com/solid/solid-oidc-tests
  * ...: It uses github pages to produce a static app
  * ...: I can, for example, test Solid OIDC issuer. Then it provides a conformance report explaining which tests are passing and which are failing. It also provides MUST / MAY level.
  * ...: It would be nice to get results as RDF. All the tests are described as RDF already.
  * ...: Pete had suggestion, I use comment now, but spec vocab (currently on branch not published) has property called statement.
  * ...: There are some things hard coded; for example, using specific crypt algorithms. CSS team has run it against their server; it was only failing very recent change to the spec introducing WebID scope.
  * ...: Some UI design could happen here, for now it is a decent start.
* eP: does it test interaction of OP with a client, or also with RS?
* AC: currently it tests OP client interaction
  * ...: it does test some OIDC interaction, well-known, public key, etc.
  * ...: I'm only testing ID Token, anticipating changes coming next
* eP: noticing annotations, there is nothing stopping us using JSON-LD in the spec document (rather than Turtle). What is the process of extracting the 'statement'.
* AC: the current test suit has no dependencies. I'm pulling descriptions as JSON, since they are in compacted JSON-LD.
  * ...: I have multiple suites. The test is just few lines of code. We pass identifier for relevant test, and whether it passed, plus optional message.
  * ...: In data directory, I can describe new test or requirement, and reference it from test suite.
  * ...: I'm performing some reflection to get methods of the class, so if you add a method to the class, it automatically runs.
* PE: When we interpret the spec, you can see requirement level and requirement subject. If you do that in turtle fragments you would have duplication.
  * ...: Is you report function going directly to HTML? I assume you can start outputting EARL.
* AC: I currently do some DOM manipulation, but we could output the data in any other way.
* PE: ??? [completely missed comment should probably be deleted, unless @edwardsph remembers what they said here]
* AC: In test suite, some requirement statements are pointing to different specs. We can't expect OIDC spec to include RDF with requirements. We can either include requirements for those specs or not test things not defined in OIDC spec.
  * ...: I think it's helpful for developers to test dependencies from OIDC specs. For example, test whether issuer matches.
  * ...: I don't know how far we can go with expecting data to be automatically loaded from the spec. Sometimes we need to have a separate manifest.
* PE: I see value for the developer of tests guiding them.
* eP: ??? [completely missed comment should probably be deleted, unless @elf-pavlik remembers what they said here]
* AC: With OIDC we point to the same IRI; we have similar issue with Solid-OIDC. For example, we have basic assumption that we use PKCE and Authorization Code flow, but it is not a normative statement. In my test, MUST is not reflecting exactly what the spec test says. We should adjust spec text to reflect those requirements.
  * ...: For discovery, we point to the same IRI when we point to WebID scope, and supported claims.
* AC: I have all of the code written as ESM. I'm receptive to suggestions for how to make code better.
```

ACTION: AC I'll share solid-oidc-tests in relevant gitter channels.

---

ACTION: eP to move PR #18 forward to be ready to merge by the end of Nov 2021

> * BR: there may be additional conversations w.r.t interop panel issues. Drawing lines between panels is getting complicated

Reviewer 2: ```suggestion
* BR: there may be additional conversations with regard to interop panel issues. Drawing lines between panels is getting complicated
```
